src/page.rs

```diff
@@ -18,12 +21,18 @@ fn escape(s: &str) -> String {
     s.replace('&', "\\u0026")
 }
 
+
+
+
 impl PageSpec {
+    //fn get_og_scripts(&self) -> String{
+    //    I have no idea where to define this function
+    //}
     pub fn render(&self, is_crawler: bool) -> Result<Vec<u8>, failure::Error> {
         let data = escape(serde_json::to_string_pretty(&self)?.as_str());
         let mut html = HTML_PAGE.clone();
 
-        html = html.replace("__realm_title__", &self.title);
+        html = html.replace("__realm_title__", &self.title);// add __og_script__, use get_og_scripts function()
```

Comment:
Write more commented out actual code so we know what parameters are being passed etc. Try to detail out as much without writing function bodies (tho write function body in comments).